src/util/triplea/MapXMLCreator/MapXMLHelper.java

```diff
@@ -0,0 +1,1008 @@
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NamedNodeMap;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import util.image.FileOpen;
+import util.image.FileSave;
+import util.triplea.MapXMLCreator.TerritoryDefinitionDialog.DEFINITION;
+
+import com.sun.org.apache.xerces.internal.impl.xpath.regex.ParseException;
+
+/**
+ * This class reads, writes and keeps the Map XML properties.
+ * 
+ * @author Erik von der Osten
+ * 
+ */
+public class MapXMLHelper {
+
+	private static final String TRIPLEA_JAVA_CLASS_DELEGATE_PATH = "games.strategy.triplea.delegate.";
+	static LinkedHashMap<String, String> s_xmlStrings = new LinkedHashMap<String, String>();
+	static ArrayList<String> s_resourceList = new ArrayList<String>();
+	static HashMap<String, HashMap<DEFINITION, Boolean>> s_territoyDefintions = new HashMap<String, HashMap<DEFINITION, Boolean>>();
+	static HashMap<String, LinkedHashSet<String>> s_territoyConnections = new HashMap<String, LinkedHashSet<String>>();
+	static ArrayList<String> s_playerName = new ArrayList<String>();
+	static HashMap<String, String> s_playerAlliance = new HashMap<String, String>();
+	static HashMap<String, Integer> s_playerInitResources = new HashMap<String, Integer>();
+	static LinkedHashMap<String, ArrayList<Integer>> s_unitDefinitions = new LinkedHashMap<String, ArrayList<Integer>>();
+	static LinkedHashMap<String, ArrayList<String>> s_gamePlaySequence = new LinkedHashMap<String, ArrayList<String>>();
+	static LinkedHashMap<String, Triple<String, String, Integer>> s_playerSequence = new LinkedHashMap<String, Triple<String, String, Integer>>();
+	static LinkedHashMap<String, ArrayList<String>> s_technologyDefinitions = new LinkedHashMap<String, ArrayList<String>>();
+	static LinkedHashMap<String, ArrayList<String>> s_productionFrontiers = new LinkedHashMap<String, ArrayList<String>>();
+	static LinkedHashMap<String, ArrayList<String>> s_unitAttatchments = new LinkedHashMap<String, ArrayList<String>>();
+	static HashMap<String, Integer> s_territoyProductions = new HashMap<String, Integer>();
+	static HashMap<String, Tuple<SortedSet<String>, SortedSet<String>>> s_canalDefinitions = new HashMap<String, Tuple<SortedSet<String>, SortedSet<String>>>();
+	static HashMap<String, String> s_territoyOwnerships = new HashMap<String, String>();
+	static HashMap<String, HashMap<String, LinkedHashMap<String, Integer>>> s_unitPlacements = new HashMap<String, HashMap<String, LinkedHashMap<String, Integer>>>();
+	static HashMap<String, ArrayList<String>> s_gameSettings = new HashMap<String, ArrayList<String>>();
+	static String s_notes = "";
+
+	static File s_mapXMLFile;
+
+	static int loadXML() {
+		try {
+			final String gameXMLPath = new FileOpen("Load a Game XML File", s_mapXMLFile, ".xml").getPathString();
+			if (gameXMLPath == null) {

```

Comment:
please change gameXmlPath to "Optional< .. >", then initialize it to "Optional.empty()" instead of null. Then you'll avoid the chance of having an NPE. So this code would change to:
```
if ( !gameXmlPath.isPresent() ) {
  return -1;
}